dvc/repo/__init__.py

```diff
@@ -455,9 +456,15 @@ def open(self, path, remote=None, mode="r", encoding=None):
         if out.isdir():
             raise ValueError("Can't open a dir")
 
+        cache_file = self.cache.local.checksum_to_path_info(out.checksum)
+        cache_file = fspath_py35(cache_file)
+
         with self.state:
             cache_info = out.get_used_cache(remote=remote)
             self.cloud.pull(cache_info, remote=remote)
+            if not os.path.exists(cache_file):
+                raise DvcException(
+                    "Can't find {path} neither locally nor on remote"
```

Comment:
are we testing this?

missing `.format(path` ...?

Reply:
what is the motivation to throw this here instead of relying on the `_open` itself? To unify the behavior across Pythons?

Reply:
also, does it make sense to make a specific exception `DvcFileNotFoundException`? And put more info into the exception message - like remote info we tried to pull it from.

Reply:
When built-in `open()` raises here the error message looks quite cryptic, it is some random cache path not existing, while the usual real reason is it absent on remote. `self.cloud.pull()` already shows a warning with what is not found, so no need to add something about it here. 

I will write a test.